pandas/tests/frame/indexing/test_mask.py

```diff
@@ -130,3 +131,12 @@ def test_mask_where_dtype_timedelta():
         [np.nan, np.nan, np.nan, Timedelta("3 day"), Timedelta("4 day")]
     )
     tm.assert_frame_equal(df.where(df > Timedelta(2, unit="d")), expected)
+
+
+def test_mask_return_dtype():
+    # GH#50488
+    ser = Series([0.0, 1.0, 2.0, 3.0], dtype=Float64Dtype)
+    cond = ~ser.isna()
+    other = Series([True, False, True, False])
+    excepted = Series([1.0, 0.0, 1.0, 0.0], dtype=ser.dtype)
+    tm.assert_series_equal(ser.mask(cond, other), excepted)
```

Comment:
Can you define result in one extra line? Makes it easier to read

Reply:
like this?
```python
result = ser.mask(cond, other)
tm.assert_series_equal(result, excepted)
```

Reply:
Yep